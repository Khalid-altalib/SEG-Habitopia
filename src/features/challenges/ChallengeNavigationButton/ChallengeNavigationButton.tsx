import { useNavigation } from "@react-navigation/native";
import React from "react";
import { ButtonType, NavigationParams } from "types";
import { NavigationProp } from "@react-navigation/native";
import Button from "@components/Button";

type Props = {};

function ChallengeNavigationButton({}: Props) {
  const navigation = useNavigation<NavigationProp<NavigationParams>>();
  const handlePress = () => navigation.navigate("Explore");
  return (
    <Button
      type={ButtonType.Primary}
      isFullWidth
      style={{ marginHorizontal: 25, marginBottom: 25, paddingTop: 10 }}
      onPress={handlePress}
    >
      Discover more!
    </Button>
  );
}

export default ChallengeNavigationButton;
