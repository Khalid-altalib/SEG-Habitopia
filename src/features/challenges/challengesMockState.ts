import { ChallengesState } from "./challengesSlice";

// Mocks the challenge state for testing purposes
const challengesMockState: ChallengesState = {
    challenges: [
        {
            id: '1',
            name: 'Fitness',
            description: 'Be active',
            active: true

        },
        {
            id: '1',
            name: 'Study',
            description: 'Study hard',
            active: false
        }
    ],
    fetchChallenges: {
        loading: false,
        error: ""
    },
    joinChallenge: {
        loading: false,
        error: ""
    }
};

export default challengesMockState;