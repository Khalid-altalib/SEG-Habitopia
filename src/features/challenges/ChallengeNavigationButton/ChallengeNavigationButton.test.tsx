import { fireEvent, render, screen } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import challengesMockState from "../challengesMockState";
import { extendTheme, NativeBaseProvider } from "native-base";
import ChallengeNavigationButton from "./ChallengeNavigationButton";

describe("ChallengeBoxes", () => {
  const mockState = {
    challenges: challengesMockState,
  };
  const mockStore = configureStore([thunk])(mockState);

  it("renders the challenge boxes correctly", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <ChallengeNavigationButton />
      </TestingWrapperNavigation>
    );

    expect(component).toMatchSnapshot();
  });

  it("navigates to the explore page", async () => {
    const theme = extendTheme({});
    const inset = {
      frame: { x: 0, y: 0, width: 0, height: 0 },
      insets: { top: 0, left: 0, right: 0, bottom: 0 },
    };
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
          <ChallengeNavigationButton />
        </NativeBaseProvider>
      </TestingWrapperNavigation>
    );
    const button = component.getByText("Discover more!");

    fireEvent.press(button);

    const check = screen.queryByText("Discover more!");
    expect(check).toBeNull();
  });
});
