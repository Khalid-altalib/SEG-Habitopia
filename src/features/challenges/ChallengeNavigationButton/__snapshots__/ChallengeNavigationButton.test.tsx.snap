// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeBoxes renders the challenge boxes correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              isFullWidth={true}
              onPress={[Function]}
              style={
                {
                  "marginBottom": 25,
                  "marginHorizontal": 25,
                }
              }
              type={0}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "marginBottom": 25,
                    "marginHorizontal": 25,
                    "opacity": 1,
                  }
                }
              >
                <ViewManagerAdapter_ExpoLinearGradient
                  proxiedProperties={
                    {
                      "colors": [
                        4278220774,
                        4286456526,
                      ],
                      "endPoint": [
                        1,
                        1,
                      ],
                      "locations": undefined,
                      "startPoint": [
                        0,
                        0,
                      ],
                    }
                  }
                  style={
                    [
                      {
                        "alignItems": "center",
                        "borderRadius": 10,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingHorizontal": 40,
                        "paddingVertical": 15,
                      },
                      {
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#18181b",
                        },
                        {
                          "color": "#e4e4e7",
                          "fontFamily": undefined,
                          "fontSize": 19,
                        },
                        [
                          null,
                          {
                            "alignSelf": "center",
                          },
                        ],
                      ]
                    }
                  >
                    Discover more!
                  </Text>
                  <View
                    style={
                      {
                        "marginLeft": 5,
                      }
                    }
                  />
                </ViewManagerAdapter_ExpoLinearGradient>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
